Rich text editor element

jonatanschroeder opened the pull request:
Resolves #4030 .

jonatanschroeder commented:
I have reviewed and hereby sign the CLA

firasm commented:
Thanks for doing this PR @jonatanschroeder ! I think this will be just what I need for some descriptive/reflective questions!

jonatanschroeder commented on exampleCourse/questions/element/richTextEditor/question.html:
```diff
@@ -0,0 +1,93 @@
+</pl-question-panel>
+
+<pl-hide-in-panel answer="true">
+  <div class="card my-2">
```
There are other questions that use this kind of construct to separate different parts:
https://github.com/PrairieLearn/PrairieLearn/blob/001eb649a53b2d623e938fa1acf51248cfa4ac11/exampleCourse/questions/element/checkbox/question.html#L12
https://github.com/PrairieLearn/PrairieLearn/blob/001eb649a53b2d623e938fa1acf51248cfa4ac11/exampleCourse/questions/element/integerInput/question.html#L10
https://github.com/PrairieLearn/PrairieLearn/blob/001eb649a53b2d623e938fa1acf51248cfa4ac11/exampleCourse/questions/element/fileEditor/question.html#L14
https://github.com/PrairieLearn/PrairieLearn/blob/001eb649a53b2d623e938fa1acf51248cfa4ac11/exampleCourse/questions/element/multipleChoice/question.html#L11

jonatanschroeder replied:
Do you prefer cards with no headers?

jonatanschroeder replied:
I replaced it with `hr` and a margin, does it look good enough?

mwest1066 replied:
Looks good!

jonatanschroeder commented on exampleCourse/questions/element/richTextEditor/question.html:
```diff
@@ -0,0 +1,93 @@
+        </p>
+      </pl-question-panel>
+      
+      <pl-rich-text-editor file-name="meaning.html" >
```
Ok, known issue with Quill's paste function. Fixed.

jonatanschroeder commented on docs/elements.md:
```diff
@@ -780,6 +782,42 @@ In the submission panel, a `pl-matrix-input` element displays either the submitt
 
 -----
 
+### `pl-rich-text-editor` element
```
Assuming you're looking into `pl-matching` (#3670) at some point in the near future, I wonder if it may be better to have a single news item with recent additions to elements. Maybe also include the `feedback` argument in `pl-multiple-choice` (#4656). And (if you're open to review them) `pl-xss-safe` in #3475 and `pl-feedback` in #3865 ?

mwest1066 replied:
Yes, sorry, I feel really bad about holding up all these other excellent element improvements for no good reason. I'm reviewing now!

mwest1066 replied:
(Feel free to merge this PR whenever you like.)

jonatanschroeder commented:
@mwest1066 do you think the styling of the element may be a bit too bland? As in the element seems to close to the text itself and students may not notice the element there? Should I maybe add something like a background or shadow borders?

jonatanschroeder commented:
Added a shadow, added an outline on focus (similar to the one for bootstrap inputs), and changed the font to use the one from the context (bootstrap itself) instead of a custom one from Quill (Helvetica/Arial).

![image](https://user-images.githubusercontent.com/1004907/131377614-255f2be3-ee09-431c-8d9a-cb385817bb63.png)

mwest1066 commented:
> @mwest1066 do you think the styling of the element may be a bit too bland? As in the element seems to close to the text itself and students may not notice the element there? Should I maybe add something like a background or shadow borders?

I agree that the styling didn't stand out enough from the question. I noticed this myself originally but I didn't have any good ideas! The drop shadow definitely helps with this.
